WcaOnRails/app/webpacker/components/Auth/Signup/ClaimWcaId.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Input, Form, Message, FormGroup, FormField,
+} from 'semantic-ui-react';
+import DatePicker from 'react-datepicker';
+import { QueryClient, QueryClientProvider, useQuery } from '@tanstack/react-query';
+import WcaSearch from '../../SearchWidget/WcaSearch';
+import I18n from '../../../lib/i18n';
+import 'react-datepicker/dist/react-datepicker.css';
+import {
+  apiV0Urls, delegatesPageUrl, personUrl, contactDobUrl,
+} from '../../../lib/requests/routes.js.erb';
+import Loading from '../../Requests/Loading';
+import { fetchJsonOrError } from '../../../lib/requests/fetchWithAuthenticityToken';
+import I18nHTMLTranslate from '../../I18nHTMLTranslate';
+import UserBadge from '../../UserBadge';
+
+const dateFormat = 'YYYY-MM-DD';
+const queryClient = new QueryClient();
+
+function ClaimWcaId() {
+  const [unconfirmedPerson, setUnconfirmedPerson] = useState();
+  const [dobVerification, setDobVerification] = useState();
+  const [delegateToVerify, setDelegateToVerify] = useState();
+  const [manualDelegateSelection, setManualDelegateSelection] = useState(false);
+  const unconfirmedWcaId = unconfirmedPerson?.item?.wca_id || '';
+  const {
+    isLoading: isCanClaimLoading,
+    data: canClaimData,
+    refetch: refetchCanClaim,
+  } = useQuery(
+    {
+      queryKey: ['canClaim', unconfirmedWcaId],
+      queryFn: () => fetchJsonOrError(apiV0Urls.persons.canClaim(unconfirmedWcaId)),
+      enabled: !!unconfirmedWcaId,
+    },
+  );
+  const {
+    isLoading: isLikelyDelegatesLoading,
+    data: likelyDelegatesData,
+    refetch: refetchLikelyDelegates,
+  } = useQuery(
+    {
+      queryKey: ['likelyDelegates', unconfirmedWcaId],
+      queryFn: () => fetchJsonOrError(apiV0Urls.persons.likelyDelegates(unconfirmedWcaId)),
+      enabled: !!unconfirmedWcaId,
+    },
+  );
+  const { data: { can_claim: canClaim, reason } } = canClaimData || { data: {} };
+  const { data: { likely_delegates: likelyDelegates } } = likelyDelegatesData || { data: {} };
+  const loading = isCanClaimLoading || isLikelyDelegatesLoading;
+
+  useEffect(() => {
+    if (unconfirmedWcaId) {
```

Comment:
I think this should be wrapped around a debounce?

Reply:
That was a good catch. I didn't think of this case. I assumed this will cause useEffect to get called in every re-render as the value is getting changed. But to my surprise it's working fine - ie, it's getting changed only when the unconfirmedWcaId is changed. I think it's because `unconfirmedPerson` is a state and hence value of `unconfirmedWcaId` is same in re-render because it's an object of `unconfirmedPerson`. Hence, I assume there is no change needed as of now. Please let me know if I missed something.